app/uk/gov/hmrc/servicedependencies/controller/ServiceDependenciesController.scala

```diff
@@ -147,26 +147,49 @@ class ServiceDependenciesController @Inject()(
   def moduleDependencies(repositoryName: String, versionOption: Option[String]): Action[AnyContent] =
     Action.async {
       for {
-        metaArtefacts  <- versionOption match {
-                            case Some(version) if version equalsIgnoreCase("latest") =>
+        latestVersions <- latestVersionRepository.getAllEntries
+        bobbyRules     <- serviceConfigsConnector.getBobbyRules
+        repositories   <- versionOption match {
+                            case Some(version) =>
                                                   metaArtefactRepository
                                                     .find(repositoryName)
-                                                    .map(_.toSeq)
-                            case Some(version) => metaArtefactRepository
-                                                    .find(repositoryName, Version(version))
-                                                    .map(_.toSeq)
+                                                    .flatMap(_ match {
+                                                      case Some(version) => Future.successful(toRepository(version, latestVersions, bobbyRules))
+                                                      case None => fallbackToDataFromCuratedSlugDependencies(repositoryName, version)
+                                                    })
+                                                    .map(Seq(_))
                             case None          => metaArtefactRepository
                                                     .findAllVersions(repositoryName)
+                                                    .map { metaArtefacts => metaArtefacts.map { meta => toRepository(meta, latestVersions, bobbyRules) }}
```

Comment:
use of underscore might remove some of the noise `.map(_.map(toRepository(_, latestVersions, bobbyRules)))`

Reply:
Done